lib/Ix/DBIC/ResultSet.pm

```diff
@@ -575,6 +575,10 @@ sub _ix_check_user_properties (
       }
     }
 
+    if (my $canonicalizer = $info->{canonicalizer}) {
+      $value = $canonicalizer->($value) if defined $value;
```

Comment:
Hrm, I was thinking something like undef => 0, or undef => '', since an explicit undef is different than not providing the field at all isn't it?

It's all hypothetical, so if you don't think it's worth it that's fine with me.